### PR TITLE
Zachmu/fix clone empty

### LINKED ISSUE #217: cloning an empty repository results in panic
After creating an empty repository on `dolthub`, attempting to clone it results in this panic:
`panic: nil commit spec`.

Here is my config:
```
metrics.host = localhost
remotes.default_host = <dev url>
user.creds = <creds>
user.email = <my email>
user.name = <my username>
```

Here is the command I ran:
`dolt clone https://doltremoteapi.dolthub.com/coffeegoddd/crashes`

### PR DESCRIPTION
**Pull request by zachmu**
Cloning an empty repo now works as expected. Fixes #217

### CODE REVIEW

File: go/cmd/dolt/commands/clone.go [hunk 1]
```diff
@@ -301,58 +301,80 @@ func cloneRemote(ctx context.Context, srcDB *doltdb.DoltDB, remoteName, branch s
...
+
+		err = dEnv.InitDBWithTime(ctx, types.Format_Default, *name, *email, doltdb.CommitNowFunc())
+		if err != nil {
+			return errhand.BuildDError("error: could not initialize repository").AddCause(err).Build()
+		}
+
+		branch = doltdb.MasterBranch
+		performPull = false
+	}
+
```
Comment (bheni) [inline go/cmd/dolt/commands/clone.go]: Maybe put this all in a function initIfEmpty or something like that with your comment at the top.

File: go/cmd/dolt/dolt.go [hunk 2]
```diff
@@ -136,7 +136,7 @@ func runMain() int {
 	restoreIO := cli.InitIO()
 	defer restoreIO()
 
-	warnIfMaxFilesTooLow()
```
Comment (bheni) [inline go/cmd/dolt/dolt.go]: Why are we commenting this out?

Comment (zachmu) [reply go/cmd/dolt/dolt.go]: For some reason this fail in GoLand for me so I comment it out locally. Reverted.

File: go/cmd/dolt/commands/clone.go (hunk 1 above)
Comment (zachmu) [reply go/cmd/dolt/commands/clone.go]: Done.

**Pull request merged**